Neos.Neos/Classes/Domain/NodeLabel/NodeLabelGeneratorInterface.php

```diff
@@ -12,20 +12,18 @@
 
 declare(strict_types=1);
 
-namespace Neos\ContentRepository\Core\NodeType;
+namespace Neos\Neos\Domain\NodeLabel;
 
 use Neos\ContentRepository\Core\Projection\ContentGraph\Node;
 
 /**
- * Interface for rendering a node label string based on some strategy
- *
- * @api
+ * @api to access the Node's label in PHP, in Fusion one can use ${q(node).label()}.
  */
 interface NodeLabelGeneratorInterface
 {
     /**
      * Render a node label
      * @api
      */
-    public function getLabel(Node $node): string;
+    public function getLabel(Node $node): NodeLabel;
```

Comment:
vs `public function renderNodeLabel(Node $node): NodeLabel;` as naming ...

Reply:
Seems fine as it is, doesn't need to be any longer

Reply:
yes and `get` while not being _correct_ ist b/c compatible :)

Reply:
Speaking of b/c: Do we really need the Value Object here?
I mean: I like it ofc but is it worth to break b/c for it?

Reply:
Also where would you hand it to? I get VOs that are used as arguments, but this is a string that in all likelyhood will be output after this, the NodeLabel will not be an argument to another method where type safety is good?

Reply:
idk, b/c is already broken due to all the namespace adjustments. Its a relatively small burden to adjust it by using `NodeLabel::fromString` ... about if its really _the_ best idea idk ... thought we would do this now and prefer more value objects. For fusion `q(node).label` will autocast it to string.

Reply:
As discussed https://neos-project.slack.com/archives/C04PYL8H3/p1715673714076229 dropping it again :)